celext/lookups.go

```diff
@@ -42,7 +42,8 @@ func ProtoFieldToCELType(fieldDesc protoreflect.FieldDescriptor, generic, forIte
 		}
 	}
 
-	if fieldDesc.Kind() == protoreflect.MessageKind {
+	if fieldDesc.Kind() == protoreflect.MessageKind ||
+		fieldDesc.Kind() == protoreflect.GroupKind {
```

Comment:
If this fixes behavior with edition 2023 message fields with `features.message_encoding = DELIMITED`, this also fixes the same issue with proto2 groups.

Proto2 groups are really only syntactic sugar. When compiled to descriptors, the keyword synthesizes a message for the group, and adds a field with TYPE_GROUP that references the message. When serialized, the field is delimited by start end end tags instead of a length-delimited tag.

The edition message_encoding feature is intended to reproduce the wire representation, but in descriptors, it's represented as a regular message field with TYPE_MESSAGE, and the feature DELIMITED in an option.

`protoreflect` however aims to have a single representation, and translates the DELIMITED option back to field kind group. So as long as we treat TYPE_GROUP the same as TYPE_MESSAGE, we should be good with both proto2 groups and edition DELIMITED (assuming we don't do any serialization).

Reply:
Yeah, I had the same feeling, this should definitely fix proto2 groups. I assume we don't have any tests for proto2 groups currently, but we could always add some if desired.

Either way, I definitely have an appreciation for how clever the 2023 edition features are. It seems like if you were using official protobuf reflection libraries and supporting all of the proto2 and proto3 features it "just works", so fixing editions support is really just fixing support for those features.